scio-core/src/main/java/com/spotify/scio/transforms/BaseAsyncLookupDoFn.java

```diff
@@ -131,30 +147,65 @@ public void processElement(ProcessContext c, BoundedWindow window) {
     }
 
     final UUID uuid = UUID.randomUUID();
```

Comment:
Just a thought: If there was a way to deterministically derive a `UUID` from `input`, `inFlightRequests` would not be required as `futures` would be enough to de-duplicate.

Reply:
There's probably no need for an `UUID` so a simple hash function would do the trick.

Reply:
You mean like uuid3/uuid5, right? Won't we still need to close over the timestamp & window for this particular input?

We'd still need to emit the output multiple times... I guess in that case we could still do the logic in lines 150-178 but with a new random uuid for the deduplicated request, accepting the very low chance of collision between our random uuid and the deterministic ones